docs/docs/ref-03-component-specs.md

```diff
@@ -37,6 +37,10 @@ object getInitialState()
 
 Invoked once before the component is mounted. The return value will be used as the initial value of `this.state`.
 
+> Note:
+>
+> This method is not available on ES6 `class` components that extend `React.Component`. It may be removed entirely in a future version of React.
```

Comment:
If you need to do something complicated, yes. You can do basic setting based on props with initializers.

```
class MyComponent extends React.Component {
  state = {
    foo: this.props.bar,
  }

  render() {
    return (
      <div>{this.state.foo}</div>
    );
  }
}
```